utils/color_converter.py

```diff
@@ -1,442 +1,52 @@
-        self.as_rgba = name_to_rgba(name)
-
-    def _from_hex(self, hex_str: str):
-        self.as_name = hex_to_name(hex_str)
-        self.as_hex = hex_str
-        self.as_rgb = rgb_to_name(self.as_name)
-        self.as_rgba = name_to_rgba(self.as_name)
-
-    def _from_rgb(self, rgb_str: str):
-        self.as_name = rgb_to_name(rgb_str)
-        self.as_hex = name_to_hex(self.as_name)
-        self.as_rgb = rgb_str
-        self.as_rgba = name_to_rgba(self.as_name)
-
-    def _from_rgba(self, rgba_str: str):
-        # Split string into r, g, b, and a
-        rgba_parts = rgba_str.lower().replace(" ", "").removeprefix("rgba(").removesuffix(")").split(",")
-
-        # Cut off trailing 0's
-        rgba_parts[-1] = str(float(rgba_parts[-1]))
-
-        rgba_str = ", ".join(rgba_parts)
-
-        self.as_name = rgba_to_name(rgba_str)
-        self.as_hex = name_to_hex(self.as_name)
-        self.as_rgb = name_to_rgb(self.as_name)
-        self.as_rgba = rgba_str
+from colour import Color as Col
+
+
+class Color(Col):
+
+    def __init__(self, val: str):
+        val = val.replace(" ", "")
+
+        def parse_float(s: str):  # convert any number to float in range [0,1]
+            if s.endswith("%"):
+                return float(s[:-1]) / 100
+            elif int(s[0]) > 0:
+                return float(s) / 255
+            else:
+                return float(s)
+
+        def parse_triple(s: str):
+            ls = s[1:-1].split(",")
+            if len(ls) != 3:
+                raise ValueError()
+            return tuple(parse_float(x) for x in ls[:3])
+
+        def parse_quadruple(s: str):
+            ls = s[1:-1].split(",")
+            if len(ls) != 4:
+                # not a quadruple, try parsing as a triple with alpha == 1
+                return parse_triple(s), 1.0
+            return tuple(parse_float(x) for x in ls[:3]), float(ls[3])
+
+        if val.startswith("#"):
+            if len(val[1:]) == 3 or len(val[1:]) == 6:  # hex is in web format
+                super(Color, self).__init__(val)
+            elif len(val[1:]) == 4:
+                alpha: int = int(val[-1:], 16)
+                super(Color, self).__init__(val[:-1])
+            elif len(val[1:]) == 8:
+                alpha: int = int(val[-2:], 16)
+                super(Color, self).__init__(val[:-2])
+        elif val.startswith("rgba"):
+            triple, alpha = parse_quadruple(val[4:])
+            super(Color, self).__init__(rgb=triple)
+        elif val.startswith("rgb"):
+            triple = parse_triple(val[3:])
+            super(Color, self).__init__(rgb=triple)
+        elif val.startswith("hsla"):
+            triple, alpha = parse_quadruple(val[4:])
+            super(Color, self).__init__(hsl=triple)
+        elif val.startswith("hsl"):
+            triple = parse_triple(val[3:])
+            super(Color, self).__init__(hsl=triple)
+        elif val[0].isalpha():
+            super(Color, self).__init__(val)
```

Comment:
Why is `alpha` never used in any of these cases? Don't we still have to store the value somewhere to properly compare `rgba` & `hsla` colours?

Reply:
forgot about that one, thanks